packages/plans-grid-next/src/hooks/data-store/use-grid-plans-for-comparison-grid.ts

```diff
@@ -6,10 +6,26 @@ import { useMemo } from '@wordpress/element';
 import useGridPlans from './use-grid-plans';
 import useRestructuredPlanFeaturesForComparisonGrid from './use-restructured-plan-features-for-comparison-grid';
 import type { UseGridPlansParams } from './types';
-import type { GridPlan } from '../../types';
+import type { GridPlan, PlansIntent } from '../../types';
 
 const HIDDEN_PLANS = [ PLAN_HOSTING_TRIAL_MONTHLY, PLAN_ENTERPRISE_GRID_WPCOM ];
 
+const getIntentForComparisonTable = ( intent: PlansIntent ): PlansIntent => {
+	if (
+		[
+			'plans-guided-segment-merchant',
+			'plans-guided-segment-blogger',
+			'plans-guided-segment-nonprofit',
+			'plans-guided-segment-consumer-or-business',
+			'plans-guided-segment-developer-or-agency',
+		].includes( intent ?? '' )
+	) {
+		return 'plans-default-wpcom';
+	}
```

Comment:
To avoid creating the array and running `includes`, we can do this:

```suggestion
	if (
		intent && [
			'plans-guided-segment-merchant',
			'plans-guided-segment-blogger',
			'plans-guided-segment-nonprofit',
			'plans-guided-segment-consumer-or-business',
			'plans-guided-segment-developer-or-agency',
		].includes( intent )
	) {
		return 'plans-default-wpcom';
	}
```